source/attriCoord.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef ATTRICOORD_HPP
+#define ATTRICOORD_HPP
+
+#include <iostream>
+#include "vector.hpp"
+
+namespace cpp_utils {
+
+class AttriCoord{
+    public:
+        double x, y, z, a, b;
```

Comment:
would this work better as the x, y and z actually being a vector then the class having and A and  B attribute as well, Should simplify most of the operators